test/e2e/volumes.go

```diff
@@ -170,6 +171,14 @@ func volumeTestCleanup(client *client.Client, config VolumeTestConfig) {
 	}
 
 	if config.serverImage != "" {
+		if err := waitForPodTerminatedInNamespace(client, config.prefix+"-client", "", config.namespace); !apierrs.IsNotFound(err) {
+			expectNoError(err, "Failed to wait client pod terminated: %v", err)
+		}
+		// See issue #24100.
+		// Prevent umount errors by making sure making sure the client pod exits cleanly *before* the volume server pod exits.
+		By("sleeping a bit so client can stop and unmount")
+		time.Sleep(20 * time.Second)
```

Comment:
20 seconds seems pretty arbitrary there are wait.Until & wait.poll which may be better here.